src/components/Experience.tsx

```diff
@@ -31,28 +26,27 @@ const experienceDetails: ExperienceDetail[] = [
 ];
 
 const Experience = () => {
-  const [currentExperienceDataIndex, updateExperienceDataIndex] = useState<
-    undefined | number
-  >(undefined);
+  const [currentExperience, setExperience] = useState<undefined | IExperience>(
+    undefined
+  );
 
   return (
     <Grid
       templateAreas={`"timeline experienceDetails"`}
       templateRows={"1fr"}
       templateColumns={"20rem 1fr"}
       height="100%"
+      p={5}
     >
-      <GridItem area="timeline" p={5}>
+      <GridItem area="timeline">
         <ExperienceTimeline
           experienceDetails={experienceDetails}
-          updateExperienceDataIndex={updateExperienceDataIndex}
+          setExperience={setExperience}
```

Comment:
By convention, I think we should name the "setExperience" callback prop as "onExperienceChange".